src/utils.c

```diff
@@ -254,3 +257,95 @@ is_scanner_only_pref (const char *pref)
     return 1;
   return 0;
 }
+
+/** @brief Writes scripts stats into a file.
+ *
+ * @param buf String to write.
+ * @param scan_id Scan ID for the file name.
+ * @param mode 2 to create the file, 0 to append text to the file,
+ *             1 to finish the json list removing the trailing comma before
+ *             appending the last text in the buffer.
+ *
+ */
+void
+write_script_stats (const char *buf, const char *scan_id, int mode)
+{
+  char *p = NULL;
+  FILE *path = NULL;
+  if (!prefs_get ("report_scripts"))
+    return;
+  p =
+    g_strdup_printf ("%s/%s-stats.json", prefs_get ("report_scripts"), scan_id);
+
+  if (mode == 0)
+    path = fopen (p, "a");
+  else if (mode == 1)
+    path = fopen (p, "r+");
+  else if (mode == 2)
+    path = fopen (p, "w");
+
+  if (path == NULL)
+    {
+      g_warning ("%s: Error opening FILE for script stats: %d - %s", __func__,
+                 errno, strerror (errno));
+      return;
+    }
+
+  if (mode == 1)
```

Comment:
This seems like debug information? 

I don't think that if should be controlled by the caller of that function. Normally you would want those information when you are either working on that part or if something is wrong. In that case I think it is error prone when you have to change the actual business logic (which I assume is never 1) to achieve that.

Reply:
Is not debug information. I forgot to removed the last two line inside the `if` statement. This statement reads to the EOF and positions the pointer in the last char. Is the way I found to remove de trailing comma from the last json object.